IdentityCore/src/network/MSIDHttpRequest.m

```diff
@@ -70,8 +71,17 @@ - (void)sendWithBlock:(MSIDHttpRequestDidCompleteBlock)completionBlock
     __auto_type requestConfigurator = [MSIDOAuthRequestConfigurator new];
     requestConfigurator.timeoutInterval = _requestTimeoutInterval;
     [requestConfigurator configure:self];
-
-    self.urlRequest = [self.requestSerializer serializeWithRequest:self.urlRequest parameters:self.parameters headers:self.headers];
+    if ([self.experimentBag msidBoolObjectForKey:MSID_EXP_ENABLE_CONNECTION_CLOSE])
+    {
+        NSMutableDictionary *localHeaders = self.headers ? [self.headers mutableCopy] : [NSMutableDictionary dictionary];
+        [localHeaders setValue:@"close" forKey:@"Connection"];
+        self.urlRequest = [self.requestSerializer serializeWithRequest:self.urlRequest parameters:self.parameters headers:[NSDictionary dictionaryWithDictionary:localHeaders]];
+    }
+    else
+    {
+        self.urlRequest = [self.requestSerializer serializeWithRequest:self.urlRequest parameters:self.parameters headers:self.headers];
```

Comment:
nit: we can refactor a bit such that we don't need to duplicate line 78 and line 82 :)